src/components/data-table/DataTable.vue

```diff
@@ -66,6 +66,46 @@
 // icons
 import { SearchIcon, SwitchVerticalIcon, SortAscendingIcon, SortDescendingIcon } from '@heroicons/vue/outline'
 
+function searchObjectProps (object, searchTerm, searchProps = []) {
+    const searchPropsMap = searchProps
+        .map((prop) => {
+            const [first, ...rest] = prop.split('.')
+
+            return [first, rest.join('.')]
+        })
+        .reduce((map, [propName, subProp]) => {
+            if (!map.has(propName)) {
+                map.set(propName, [subProp])
+            } else {
+                map.get(propName).push(subProp)
+            }
+
+            return map
+        }, new Map())
```

Comment:
Swapped to a `Map()` to decrease lookup time and avoid having to loop over the values further down.